### DISCUSSION TITLE
How to change solenoid chromatin fiber model configuration

### DISCUSSION THREAD

**Discussion by Geoffvinc** [topas-nbio/TOPAS-nBio#19] [How to?]
Hello, I wonder if I can change the configuration of the solenoid chromatin fiber model in TOPAS-nBio. The default configuration is shown below:
![image](https://github.com/topas-nbio/TOPAS-nBio/assets/116862510/944bd102-b182-4f3f-8afd-f5f283e901d6)
Is it possible to modify the number of turn of the double helix around the histone? For example, making it to 1.45 left-hand turn instead of 1.65 left-hand turn and assigning the remaining base pairs (from the 0.2 left-hand turn) to the linker DNA to simulate euchromatin (loose) structure of the chromatin.

What I have tried is I increased the radius of the fiber, But it seems that it only stretch the linker DNA, while maintaining the number of turn of the DNA around the histone. The good thing is it still maintain the same number of base pairs in the chromatin model. However, it is not something that I want to obtain.

Thank you in advance for your help!

**Comment by Geoffvinc** [comment]
Hi, I figured out a possible solution to this problem. For my case, I want to change the chromatin configuration inside a nucleus. Open TsNucleus.cc file inside 
`YOURPATH\TOPAS-nBio\geometry\nucleusModel` then you can modify the `nCoils` variable inside `GenerateDNAPath` function.

Meanwhile, combining it with increasing the chromatin fiber radius seems like a good idea to model the euchromatin structure, since the distance between nucleosome is more spaced out in this configuration.

I am still not sure on whether if changing `nCoils` would affect the chromatin base pair count. I will take a look on it and I will update here when I find out on it.

**Reply by Geoffvinc** [reply]
Hi there! I figured that changing the `nCoils` variable does not change the number of base pairs wrapping the histone. So, if you increase the `nCoils`, the same number of base pairs wrapping the histone is the same, but their distribution will be more spaced out. If you want to maintain the same distribution as the original one, you need to modify the number of segmentation (`nSteps`), which equals to the number of base pairs.

Assuming you want to maintain the total base pairs in the chromatin, you can allocate the remaining base pair to the bezier curve (the linker base pairs which connect the histone).

**Reply by JoseRamosMendez** [reply]
Does it means you find a solution to your needs?

**Reply by Geoffvinc** [reply]
Hi Jose, yes I have found the solution I need. Thank you!
